includes/class-wc-pakettikauppa-shipping-method.php

```diff
@@ -555,7 +533,7 @@ private function my_global_form_fields() {
           'show_pakettikauppa_shipping_method' => array(
             'title'   => __('Show Pakettikauppa shipping method', 'wc-pakettikauppa'),
             'type'    => 'select',
-            'default' => 'no',
+            'default' => 'yes',
```

Comment:
This can't be yes. It has to be no. Pakettikauppa shipping method is only to be used in some special cases and is not meant for everyone to be used.

Reply:
Ok. Why though? As an user, if I install a plugin and complete it's setup wizard, I would expect the plugin to work. With this set to `no`, it doesn't work without first going to Advanced settings under the plugin settings, and setting it to yes:

![image](https://user-images.githubusercontent.com/2719615/66743461-a00ee280-ee82-11e9-948e-0166c0b219c7.png)

Freshly installed and configured plugin, yet it's not possible to create shipping methods with Pakettikauppa. Am I doing something wrong?

--------

The part that I removed (below) doesn't make much sense to me. If `$this->instance_id` is 0 (which it pretty much always is, correct me if I'm wrong), and `show_pakettikauppa_shipping_method` setting has not been set, it gets all zones and loops through their shipping methods, and if any of them are instances of `WC_Pakettikauppa_Shipping_Method`, it will set the `show_pakettikauppa_shipping_method` setting to `yes`, otherwise `no`, and save the value in the database.

```php
if ( $this->instance_id === 0 ) {
  if ( ! $show_method_set ) {
    $shipping_zones = WC_Shipping_Zones::get_zones();
    $show_pakettikauppa_shipping_method = 'no';

    foreach ( $shipping_zones as $shipping_zone ) {
      foreach ( $shipping_zone['shipping_methods'] as $shipping_object ) {
        if ( get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method' ) {
          $show_pakettikauppa_shipping_method = 'yes';
        }
      }
    }

    $this->wc_pakettikauppa_shipment->update_setting('show_pakettikauppa_shipping_method', $show_pakettikauppa_shipping_method);
    $this->wc_pakettikauppa_shipment->save_settings();
    $settings = $this->wc_pakettikauppa_shipment->get_settings();
  }
}
```

a) saving the settings will cause the `Thank you for installing WooCommerce Pakettikauppa! To get started smoothly, please open our setup wizard` notice to potentially disappear before the user has even clicked on it, that surely isn't intended?
b) `get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method'` is never true on fresh installs, and the parent if prevents it from running when the install isnt fresh

In other words, the code I removed always sets the `show_pakettikauppa_shipping_method` setting to `no`. If thats's the intended behaviour, I'd like to know why, because it doesn't make sense from the users point of view.

Reply:
instance_id is not always 0. That if is needed, otherwise it ends up in the infinity loop, because get_zones() calls the same code in the end.
--
Pakettikauppa shipping method is not needed for module to work. Pakettikauppa shipping method only is used by merchant who are using it because of historic reasons. Merchants can use any shipping method there is (flat rate for instance). Pakettikauppa is not a traditional shipping method because it is not needed for pricing the shipment, only to extend the functionality of existing shipping methods.

Reply:
"b) get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method' is never true on fresh installs, and the parent if prevents it from running when the install isnt fresh"

Then this needs some kind of a fixing :)

Reply:
Ok, thanks. I'm not too familiar with this yet, I expected it to work like a traditional shipping method. I'll try to find another way around the notice. 

I merged in some new commits from master (that weren't there on monday when I originally cloned the repo), and the setup wizard makes more sense now.

![image](https://user-images.githubusercontent.com/2719615/66746292-0e0ad800-ee8a-11e9-9264-982e4ad87e21.png)

When I removed the chunk, this screen didn't work. It listed the zones but mapping wasn't possible, as it didn't list any shipping methods, despite me having multiple shipping methods in the zone.